Managers/PostProcessMan.cpp

```diff
@@ -0,0 +1,372 @@
+#include "PostProcessMan.h"
+#include "FrameMan.h"
+#include "Scene.h"
+#include "ContentFile.h"
+#include "Matrix.h"
+
+namespace RTE {
+
+	const std::string PostProcessMan::c_ClassName = "PostProcessMan";
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void PostProcessMan::Clear() {
+		m_PostScreenEffects.clear();
+		m_PostSceneEffects.clear();
+		m_pYellowGlow = 0;
+		m_YellowGlowHash = 0;
+		m_pRedGlow = 0;
+		m_RedGlowHash = 0;
+		m_pBlueGlow = 0;
+		m_BlueGlowHash = 0;
+		for (short i = 0; i < c_MaxScreenCount; ++i) {
+			m_ScreenRelativeEffects->clear();
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	int PostProcessMan::Create() {
+		// TODO: Make more robust and load more glows!
+		ContentFile glowFile("Base.rte/Effects/Glows/YellowTiny.bmp");
+		m_pYellowGlow = glowFile.GetAsBitmap();
+		m_YellowGlowHash = glowFile.GetHash();
+		glowFile.SetDataPath("Base.rte/Effects/Glows/RedTiny.bmp");
+		m_pRedGlow = glowFile.GetAsBitmap();
+		m_RedGlowHash = glowFile.GetHash();
+		glowFile.SetDataPath("Base.rte/Effects/Glows/BlueTiny.bmp");
+		m_pBlueGlow = glowFile.GetAsBitmap();
+		m_BlueGlowHash = glowFile.GetHash();
+
+		m_pTempEffectBitmap_16 = create_bitmap_ex(32, 16, 16);
+		m_pTempEffectBitmap_32 = create_bitmap_ex(32, 32, 32);
+		m_pTempEffectBitmap_64 = create_bitmap_ex(32, 64, 64);
+		m_pTempEffectBitmap_128 = create_bitmap_ex(32, 128, 128);
+		m_pTempEffectBitmap_256 = create_bitmap_ex(32, 256, 256);
+		m_pTempEffectBitmap_512 = create_bitmap_ex(32, 512, 512);
+
+		return 0;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void PostProcessMan::PostProcess() {
+		// First copy the current 8bpp backbuffer to the 32bpp buffer; we'll add effects to it
+		blit(g_FrameMan.GetBackBuffer8(), g_FrameMan.GetBackBuffer32(), 0, 0, 0, 0, g_FrameMan.GetBackBuffer8()->w, g_FrameMan.GetBackBuffer8()->h);
+
+		// Set the screen blender mode for glows
+		set_screen_blender(128, 128, 128, 128);
+
+		//acquire_bitmap(m_pBackBuffer8);
```

Comment:
This isn't a simple getter.
https://www.allegro.cc/manual/4/api/bitmap-objects/acquire_bitmap

Since i'm not 15% sure how it works i'm keeping it there for future reference cause maybe it was correct to use it there.

Reply:
Reading said link, it's junk for us
```
Acquires the specified video bitmap prior to drawing onto it. You never need to call the function explicitly as it is low level, and will only give you a speed up if you know what you are doing. Using it wrongly may cause slowdown, or even lock up your program.

Note: You do never need to use acquire_bitmap on a memory bitmap, i.e. a normal bitmap created with create_bitmap. It will simply do nothing in that case.
```
The bitmap we're getting is created with create_bitmap_ex, so I'm pretty sure that note applies, and the rest applies too.

Reply:
One of my long term endeavors has been trying to make the framebuffers work as video bitmaps. I'm keeping these for reference.

Reply:
Hmmm, okay. It'd be better to either keep as a note somewhere for yourself, or put in a separate branch for this so it doesn't clutter things up, but also that's extra effort and it's pretty minor. Stick a TODO on it to remove it when you implement framebuffers as video bitmaps.